src/UmbracoExamine.PDF/PdfIndexValueSetBuilder.cs

```diff
@@ -0,0 +1,49 @@
+﻿using Examine;
+using System.Collections.Generic;
+using Umbraco.Core;
+using Umbraco.Core.IO;
+using Umbraco.Core.Models;
+using Umbraco.Examine;
+
+namespace UmbracoExamine.PDF
+{
+    public interface IPdfIndexValueSetBuilder : IValueSetBuilder<IMedia> { }
+
+    /// <summary>
+    /// Builds a ValueSet for PDF Documents
+    /// </summary>
+    public class PdfIndexValueSetBuilder : IPdfIndexValueSetBuilder
+    {
+        private PdfTextService _pdfTextService;
+
+        public PdfIndexValueSetBuilder(PdfTextService pdfTextService)
+        {
+            _pdfTextService = pdfTextService;
+        }
+        public IEnumerable<ValueSet> GetValueSets(params IMedia[] content)
+        {
+            foreach (var item in content)
+            {
+                var umbracoFile = item.GetValue<string>(Constants.Conventions.Media.File);
+                if (string.IsNullOrWhiteSpace(umbracoFile)) continue;
+                var filepath = IOHelper.MapPath(umbracoFile);
+                var fileTextContent = ExtractTextFromFile(filepath);
+                var indexValues = new Dictionary<string, object>
+                {
+                    ["fileTextContent"] = fileTextContent,
+                    ["__IndexType"] = IndexTypes.Media,
```

Comment:
__IndexType and __NodeId are not explicitly added here, they are added by examine based on the `ValueSet` itself (since you pass these values into the `ValueSet` ctor)

Reply:
Have added additional required fields here like "id" and "name", have also used "pdf" as the index category and use the content type alias as the item type like the other umbraco based indexes.